### PR TITLE
fix(core): fix autocomplete edge cases

### LINKED ISSUE #810: Autocomplete not working when using in shell functions or different binary name
For instance in zsh:
```zsh
test() {
  wHATEVER=foo scw "$@"
}
compdef -e 'WHATEVER=foo _scw' stg
```
breaks to autocompletion for test, which should be working.

Same thing happend if I change the binary name `scw` to another one.

### PR DESCRIPTION
**Pull request by Sh4d1**
Fixes #810 

- a test is removed since it's never used in real world: autocompletion is only called once the command in typed and a space is added

Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

**Comment by jerome-quere** [comment]
Instead of replacing hardcoded `scw` by `os.Args[0]` it could be a better idea to make `BinaryName` a parameter in the `BootstrapConfig` this config could be accessible through context where needed

File: internal/core/autocomplete.go [hunk 1]
```diff
@@ -196,7 +197,7 @@ func (node *AutoCompleteNode) isLeafCommand() bool {
 // BuildAutoCompleteTree builds the autocomplete tree from the commands, subcommands and arguments
 func BuildAutoCompleteTree(commands *Commands) *AutoCompleteNode {
 	root := NewAutoCompleteCommandNode()
-	scwCommand := root.GetChildOrCreate("scw")
+	scwCommand := root.GetChildOrCreate(os.Args[0])
```
Comment (jerome-quere) [inline internal/core/autocomplete.go:200]: Yes But... we should not realy on `os.Args[0]` as this is a global varibale.
The binary name should be a parameter in BootstrapConfig and passed allong with context,
